### PR TITLE
[MM-13470] Migrate 'components/team_members_dropdown' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by clarmso**
#### Summary

This PR convert the existing TeamMembersDropdown component and its test from Javascript to Typescript.

#### Ticket Link

https://github.com/mattermost/mattermost-server/issues/13470

#### Related Pull Requests

- Has server changes N/A
- Has redux changes N/A
- Has mobile changes N/A

#### Screenshots

N/A

### CODE REVIEW

**Comment by clarmso** [comment]
Cherry picked changes from https://github.com/mattermost/mattermost-webapp/pull/6368

File: components/team_members_dropdown/__snapshots__/team_members_dropdown.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,189 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (clarmso) [inline components/team_members_dropdown/__snapshots__/team_members_dropdown.test.tsx.snap:1]: The source of the conflict from my last PR may be from this snap file. This file was gone. (Why?)

Comment (clarmso) [inline components/team_members_dropdown/__snapshots__/team_members_dropdown.test.tsx.snap:1]: The source of the conflict from my last PR may be from this snap file. This file was gone. (Why?)

**Review by devinbinnie — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thank you @clarmso 
Tested, looks good to merge.
- Verified "Make Team Admin/Member" drop down menu on Team>Manage Members works as before.
@hmhealey Can you please help merge? Thanks 🙂

**Pull request merged**